submodules/LIB/scripts/iob_module.py

```diff
@@ -430,7 +432,7 @@ def _generate_sw(cls, mkregs_obj, reg_table):
             mkregs_obj.write_swheader(
                 reg_table, cls.build_dir + "/software/src", cls.name
             )
-        mk_conf.conf_h(cls.confs, cls.name, cls.build_dir + "/software/src")
+            mk_conf.conf_h(cls.confs, cls.name, cls.build_dir + "/software/src")
```

Comment:
> looks like iob_module changes are breaking this , is that right ?

yes, the iob-soc system does not have registers (no `cls.regs` dictionary), therefore, this change is preventing the system from generating the `iob_soc_conf.h` header.
Macros from this header are used in the IOb-SoC firmware.
For example, the peripheral numbers defined in that header are used to calculate their base addresses, with help from the `iob_soc_periphs.h` file.

The Tester system uses even more of these macros, like detecting when the system runs with INIT_MEM=1:
https://github.com/IObundle/iob-soc-sut/blob/1e503de24c18420471cc53887d7140d8499db6d2/submodules/TESTER/software/src/iob_soc_tester_firmware.c#L160
Or calculating memory addresses:
https://github.com/IObundle/iob-soc-sut/blob/1e503de24c18420471cc53887d7140d8499db6d2/submodules/TESTER/software/src/iob_soc_tester_firmware.c#L327

Reply:
Therefore I think we should revert the changes from lines 425 and 435 of `iob_module.py`
And this way, keep generating the `*_conf.h` header for systems/cores without registers.

Reply:
The problem with that is that every module will create a _conf.h file, adding dozens of unused files

Reply:
Are unused files an issue?
We used to have the `flows` attribute to prevent that, however, we decided to deprecate that attribute and now every 'flow' is set up for every core (therefore, the software files are also copied/generated for every core even if they are not used).
I'm not sure why we decided to remove the flows, but I assume it's because there was no issue with having unused files in the build directory.
The flows were removed on this PR: https://github.com/IObundle/iob-soc/pull/651

Reply:
The old `emb` flow would prevent modules from generating the `*_conf.h` file if they did not have this flow.
https://github.com/IObundle/iob-soc/pull/651/files#diff-352ff413d1b7a3a396110618b053344ae8d4ed65f727a5a38de8505151123578L407-L419